src/widgets/form.ts

```diff
@@ -0,0 +1,109 @@
+
+import {
+    DOMWidgetModel,
+    DOMWidgetView,
+    ISerializers,
+} from '@jupyter-widgets/base';
+
+import { MODULE_NAME, MODULE_VERSION } from '../version';
+
+// Import the CSS
+import '../../style/widget.css';
+
+export class FormModel extends DOMWidgetModel {
+    defaults() {
+        return {
+            ...super.defaults(),
+            _model_name: FormModel.model_name,
+            _model_module: FormModel.model_module,
+            _model_module_version: FormModel.model_module_version,
+            _view_name: FormModel.view_name,
+            _view_module: FormModel.view_module,
+            _view_module_version: FormModel.view_module_version,
+            value: 'Hello World',
+        };
+    }
+
+    static serializers: ISerializers = {
+        ...DOMWidgetModel.serializers,
+        // Add any extra serializers here
+    };
+
+    static model_name = 'FormModel';
+    static model_module = MODULE_NAME;
+    static model_module_version = MODULE_VERSION;
+    static view_name = 'FormView'; // Set to null if no view
+    static view_module = MODULE_NAME; // Set to null if no view
+    static view_module_version = MODULE_VERSION;
+}
+
+export class FormView extends DOMWidgetView {
+    render() {
+        this.el.classList.add('custom-widget');
+
+        const template = `
+        <form id="myForm">
+        <label for="dropdown">Select an option:</label>
+        <select id="dropdown" name="dropdown">
+          <option value="A">Option A</option>
+          <option value="B">Option B</option>
+        </select>
+      
+        <label for="port">Enter a port:</label>
+        <input type="number" id="port" name="port">
+
+        <div id="confirmationMessage"></div>
+      
+        <button type="submit">Submit</button>
+      </form>
+      
+`
+
+        this.el.innerHTML = template;
+
+        // Add event listener for form submission
+        const form = this.el.querySelector('#myForm');
+        form.addEventListener('submit', this.handleFormSubmit.bind(this));
+
+        // Listen for messages from the Python backend
+        this.model.on('msg:custom', this.handleMessage.bind(this));
```

Comment:
The backend we are referring to is `class FormWidget(DOMWidget)`. I think we should mention it since it may confuse if someone wants to connect to `JupySQL` backend.